app/stac_api/pgtriggers.py

```diff
@@ -61,79 +80,114 @@ def generate_child_triggers(parent_name, child_name):
-            ) a
-        GROUP BY collection_id;
+        RAISE INFO
+            '{count_table} (collection_id, value) (% %) count updated, due to asset.name=% update.',
+            related_collection_id, asset_instance.{value_field}, asset_instance.name;
 
-        -- Update related collection (auto variables + summaries)
-        UPDATE stac_api_collection SET
-            updated = now(),
-            etag = gen_random_uuid(),
-            summaries_proj_epsg = collection_summaries.proj_epsg,
-            summaries_geoadmin_variant = collection_summaries.geoadmin_variant,
-            summaries_geoadmin_lang = collection_summaries.geoadmin_lang,
-            summaries_eo_gsd = collection_summaries.eo_gsd
-        WHERE id = related_collection_id;
+        RETURN asset_instance;
+        END IF;
 
-        RAISE INFO 'collection.id=% summaries updated, due to asset.name=% update.',
-            related_collection_id, asset_instance.name;
+        RAISE INFO
+            '{count_table} (collection_id, value) (% %) deleted, due to asset.name=% update.',
+            related_collection_id, asset_instance.{value_field}, asset_instance.name;
 
         RETURN asset_instance;
         '''
 
+    class UpdateInsertCounterTrigger(pgtrigger.Trigger):
+        when = pgtrigger.After
+        declare = [('asset_instance', 'stac_api_asset%ROWTYPE'), ('related_collection_id', 'INT')]
+        func = f'''
+        asset_instance = NEW;
+
+        related_collection_id = (
+            SELECT collection_id FROM stac_api_item
+            WHERE id = asset_instance.item_id
+        );
+
+        INSERT INTO stac_api_{count_table} (collection_id, value, count)
+        VALUES (related_collection_id, asset_instance.{value_field}, 1)
+        ON CONFLICT (collection_id, value)
+        DO UPDATE SET count = stac_api_{count_table}.count+1;
+
+        RAISE INFO
+            '{count_table} (collection_id, value) (% %) count updated, due to asset.name=% update.',
+            related_collection_id, asset_instance.{value_field}, asset_instance.name;
+
+        RETURN asset_instance;
+        '''
+
+    return [
+        UpdateDeleteCounterTrigger(
+            name=f'upd_dec_{value_field}_trigger',
+            operation=pgtrigger.Update,
+            condition=pgtrigger.
+            Condition(f'''OLD.{value_field} IS DISTINCT FROM NEW.{value_field}''')
+        ),
+        UpdateDeleteCounterTrigger(
+            name=f'del_{value_field}_trigger',
+            operation=pgtrigger.Delete,
+        ),
+        UpdateInsertCounterTrigger(
+            name=f'upd_inc_{value_field}_trigger',
+            operation=pgtrigger.Update,
+            condition=pgtrigger.
+            Condition(f'''OLD.{value_field} IS DISTINCT FROM NEW.{value_field}''')
+        ),
+        UpdateInsertCounterTrigger(
+            name=f'add_{value_field}_trigger',
+            operation=pgtrigger.Insert,
+        ),
+    ]
+
+
+def generates_asset_triggers():
```

Comment:
Could this function be merged with `asset_counter_trigger` like it is with `generate_collection_asset_triggers`?